core/src/main/java/com/gentics/mesh/core/data/job/BranchMigrationJob.java

```diff
@@ -99,54 +80,54 @@ private BranchMigrationContext prepareContext() {
 
 				BranchMigrationCause cause = new BranchMigrationCause();
 				cause.setProject(newBranch.getProject().transformToReference());
-				cause.setOrigin(options().getNodeName());
+				cause.setOrigin(tx.data().options().getNodeName());
 				cause.setUuid(getUuid());
 				context.setCause(cause);
 
 				context.getStatus().commit();
 				return context;
 			});
 		} catch (Exception e) {
-			db().tx(() -> {
+			db.tx(() -> {
 				status.error(e, "Error while preparing branch migration.");
 			});
 			throw e;
 		}
 	}
 
-	@Override
-	protected Completable processTask() {
-		BranchMigration handler = mesh().branchMigrationHandler();
-
+	default Completable processTask(Database db) {
+		BranchMigration handler = db.tx(tx -> { 
```

Comment:
By the time the job is actually started, there is no transaction happening, so we have to initiate our own one.